automerge/src/lib.rs

```diff
@@ -327,11 +340,18 @@ impl Automerge {
         prop: P,
         value: i64,
     ) -> Result<OpId, AutomergeError> {
-        self.local_op(obj.into(), prop.into(), OpType::Inc(value))
+        match self.local_op(obj.into(), prop.into(), OpType::Inc(value))? {
+            Some(opid) => Ok(opid),
+            None => { panic!("increment should always create a new op") }
+        }
     }
 
     pub fn del<P: Into<Prop>>(&mut self, obj: OpId, prop: P) -> Result<OpId, AutomergeError> {
-        self.local_op(obj.into(), prop.into(), OpType::Del)
+        // TODO: Should we also no-op multiple delete operations?
```

Comment:
I think it makes sense to noop a delete if there isn't anything there to delete.

Reply:
I'll wait for Orion to weigh in as well just in case we've missed some subtleties of the successor handling logic.

Reply:
Yes.  Lets have del() that deletes nothing be a noop.  Also an increment with no matching counter.

Reply:
I think right now `inc()` will pred all visible ops - but I think it would make more sense to only pred counters.  This means that if we have conflicts with `( counter a, counter b, string c)` the increment wouldn't clear the conflict (which I like).  If we did an inc(1) we'd get `( counter a + 1, counter b + 1, string c)`  All we would need to do is make sure we only add counters to the pred and return a `None` if we find no counters.

Reply:
Is that the semantics we want? Seems like setting a key should always clear a conflict no?